nautobot/ipam/models.py

```diff
@@ -966,6 +941,39 @@ def _set_mask_length(self, value):
     mask_length = property(fset=_set_mask_length)
 
 
+class IPAddressToInterface(BaseModel):
+    ip_address = models.ForeignKey("ipam.IPAddress", on_delete=models.CASCADE)
+    interface = models.ForeignKey("dcim.Interface", blank=True, null=True, on_delete=models.CASCADE)
+    vm_interface = models.ForeignKey("virtualization.VMInterface", blank=True, null=True, on_delete=models.CASCADE)
+    source = models.BooleanField(default=False, help_text="Source address")
+    destination = models.BooleanField(default=False, help_text="Destination address")
+    default = models.BooleanField(default=False, help_text="Default address")
+    preferred = models.BooleanField(default=False, help_text="Preferred address")
+    primary = models.BooleanField(default=False, help_text="Primary address")
+    secondary = models.BooleanField(default=False, help_text="Secondary address")
+    standby = models.BooleanField(default=False, help_text="Standby address")
+    primary_for_device = models.BooleanField(default=False, help_text="Primary address for parent device")
```

Comment:
Just for my knowledge here, why do all the addresses are `BooleanFields`? Do these indicate whether `IPAddressToInterface` assignments have them?

Reply:
Here's a configuration example:

```
interface Tunnel0
 bandwidth 180
 ip address 172.17.3.3 255.255.255.0
 no ip directed-broadcast
 tunnel source 10.0.2.4
 tunnel destination 192.24.2.5
 crypto map s1first
!
```

For `Interface` Tunnel0 you would have 3 assocations:

```py
IPAddressToInterface.objects.create(
    ip_address=<IPAddress 172.17.3.3>,
    interface=<Interface Tunnel0>,
)
IPAddressToInterface.objects.create(
    ip_address=<IPAddress 10.0.2.4>,
    interface=<Interface Tunnel0>,
    source=True,
)
IPAddressToInterface.objects.create(
    ip_address=<IPAddress 192.24.2.5>,
    interface=<Interface Tunnel0>,
    destination=True,
)
```

Reply:
Maybe it's the field names and/or `help_text` that's misleading then. Should these be things like `is_default`, `is_preferred`, etc.?

Reply:
> Maybe it's the field names and/or `help_text` that's misleading then. Should these be things like `is_default`, `is_preferred`, etc.?

That sounds like a good idea to me